src/Microsoft.OData.Core/UriParser/TypePromotionUtils.cs

```diff
@@ -267,14 +267,15 @@ internal static void GetTypeFacets(IEdmTypeReference type, out int? precision, o
                     return true;
                 }
 
-                // Comparing an enum with a string is valid
-                if (left != null && right != null && left.IsEnum() && right.IsString())
+                // Comparing an enum with a string or int is valid
+                if (left != null && right != null && left.IsEnum() && (right.IsString() || right.IsIntegral()))

```

Comment:
This comment https://github.com/OData/odata.net/pull/3014/files#r1673723890 is relevant here as well. We wouldn't want to return `true` for floating point numbers

Reply:
@gathogojr 

See my previous comment:

> Floating point numbers will throw CannotConvertToType exception when used. Here is the test for this:
> 
> [IfTypePromotionNeeded_SourceIsFloatMemberValuesAndTargetIsEnum_CannotConvertToTypeExceptionIsThrown()](https://github.com/OData/odata.net/blob/fix/2542-filter-with-enum-integer/test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Binders/MetadataBindingUtilsTests.cs#L181)